framework/db/BaseActiveRecord.php

```diff
@@ -762,12 +780,11 @@ public function updateCounters($counters)
     {
         if ($this->updateAllCounters($counters, $this->getOldPrimaryKey(true)) > 0) {
             foreach ($counters as $name => $value) {
-                if (!isset($this->_attributes[$name])) {
-                    $this->_attributes[$name] = $value;
-                } else {
-                    $this->_attributes[$name] += $value;
+                if (null !== $current = $this->getAttribute($name)) {
```

Comment:
You mean `$a = $b`? We're totally covered by tests so the priority is to make it more readable.